DDD-6.md

```diff
@@ -0,0 +1,100 @@
+Perhaps more imoportant argument are the deliverables of each solutions. While Helm charts require only descriptors, with operator we wil have to also support the operator application and container. 
+
+
+Kubernetes operator provides significatly more flexibility and space for further functionality growth. For these reasons the further parts of this DDD are discussing the desing of future Debezium operator.
+
+## Operator Design & Requirements
+
+## Scope & Focus
+Do we want to create an operator for Debezium server specifically or more general Debezium operator?
+The opeator could potentially support
+
+- Debezium connector for DS
+- Debezium connector for KafkaConnect (can we somehow reuse UI backend?)
+- Debezium connector for Strimzi? Does it make sense to translate our resource to Strimzi?
+- Debezium connectors for DS via strimzi resources
+- Debezium UI deployment
+
+
+Initially will start with support for Debezium Server. However the resources should be designed with the broader scope in mind from beginning 
+
+
+### Operator SDK options
+
+The two primary options to consider her are Golang's Operator SDK (Golang) or Java Operator Framework (as Quarkus extension) . In case of a java-centered team such as our it makes sense to go with JODSK --  especially since JOSDK has quite active development. 
+
+
+### Custom Resource Proposal
+The following is a proposed example of `DebeziumServer` custom resource instance (in this). 
+
+```yaml
+apiVersion: debezium.io/v1alpha1
+kind: DebeziumServer
+metadata: 
+    name: my-debezium
+spec:
+    version: "2.2"
+    storage:
+      type: persistent 
+      claimName: ds-data-pvc
+    quarkus:
+      log.console.json: false
+    format:
+      value:
+        type: json
+        schemas.enable: false
+    transforms:
+      - name: test
+        type: com.example.TestTransform
+        predicate: test
+        negate: false
+        prop: 42
+    sink:
+      type: kafka
+      producer.bootstrap.servers: dbz-kafka-kafka-bootstrap:9092
+      producer.key.serializer: org.apache.kafka.common.serialization.StringSerializer
+      producer.value.serializer: org.apache.kafka.common.serialization.StringSerializer
+    source:
+      class: io.debezium.connector.mongodb.MongoDbConnector
+      mongodb.connection.string: mongodb://debezium:dbz@mongo.debezium.svc.cluster.local:27017/?replicaSet=rs0
+      database.history: io.debezium.relational.history.FileDatabaseHistory
+      tasks.max: 1
+      topic.prefix: demo
+      database.include.list: inventory
+      collection.include.list: inventory.customers
+      offset.storage.file.filename: /debezium/data/offsets.dat
+      offset.flush.interval.ms: 0 
+Status:
+    conditions:
+        Status: True
+        Type: Ready
+        Message: Server my-debezium is operational 
+```
+
+
+## Open Questions
+
+1) Do we need a service? What for?
```

Comment:
I don't think we need service right now. Debezium does not expose any data endpoint.

Reply:
+1 I will remove the service then